<!-- 
  Simulate Jackal robot with the teb_local_planner in gazebo:
  - gazebo including the jackal model
  - map_server
  - move_base
  - static map
  - amcl
  - rviz view

  ADDITIONAL REQUIREMENTS/DEPENDENCIES:
  - jackal_navigation
  - jackal_gazebo
 -->
<launch>

        <!--  ************** Global Parameters ***************  -->
        <param name="/use_sim_time" value="true"/>

        <!--  ************** Gazebo Simulator ***************  -->
	<include file="$(find jackal_gazebo)/launch/jackal_world.launch">
		<arg name="config" value="front_laser" />
	</include>

        <!--  ************** Navigation ***************  -->
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
  	  	<rosparam file="$(find teb_local_planner_tutorials)/cfg/jackal/costmap_common_params.yaml" command="load" ns="global_costmap" />
  	 	<rosparam file="$(find teb_local_planner_tutorials)/cfg/jackal/costmap_common_params.yaml" command="load" ns="local_costmap" />
  		<rosparam file="$(find teb_local_planner_tutorials)/cfg/jackal/local_costmap_params.yaml" command="load" />
  		<rosparam file="$(find teb_local_planner_tutorials)/cfg/jackal/global_costmap_params.yaml" command="load" />
  		<rosparam file="$(find teb_local_planner_tutorials)/cfg/jackal/teb_local_planner_params.yaml" command="load" />

		<param name="base_global_planner" value="global_planner/GlobalPlanner" />
		<param name="planner_frequency" value="1.0" />
		<param name="planner_patience" value="5.0" />

		<param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
		<param name="controller_frequency" value="5.0" />
		<param name="controller_patience" value="15.0" />

		<remap from="odom" to="odometry/filtered" />
	</node>


	<!--  ****** Maps *****  -->
	<arg name="map_file" default="$(find jackal_navigation)/maps/jackal_race.yaml"/>
 	<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

	 <!--- Run AMCL -->
 	<include file="$(find jackal_navigation)/launch/include/amcl.launch" />


	 <!--  **************** Visualisation ****************  -->
	 <node name="rviz" pkg="rviz" type="rviz" args="-d $(find teb_local_planner_tutorials)/cfg/jackal/rviz_jackal_teb.rviz"/>

</launch>
